### ISSUE TITLE
eOS 7.0 warnings

### ISSUE THREAD

**Issue by kalligator** [MvBonin/wingpanel-community-indicators#2]
As in https://github.com/MvBonin/switchboard-plug-indicators/issues/1 I get the following warnings in eOS 7
```
username@pc:~/wingpanel-community-indicators$ meson build --prefix=/usr
The Meson build system
Version: 0.61.2
Source dir: /home/username/wingpanel-community-indicators
Build dir: /home/username/wingpanel-community-indicators/build
Build type: native build
Project name: wingpanel-indicator-ayatana
Project version: undefined
C compiler for the host machine: cc (gcc 11.3.0 "cc (Ubuntu 11.3.0-1ubuntu1~22.04) 11.3.0")
C linker for the host machine: cc ld.bfd 2.38
Vala compiler for the host machine: valac (valac 0.56.3)
Host machine cpu family: x86_64
Host machine cpu: x86_64
Found pkg-config: /usr/bin/pkg-config (0.29.2)
Found CMake: /usr/bin/cmake (3.22.1)
WARNING: CMake Toolchain: Failed to determine CMake compilers state
Run-time dependency wingpanel-2.0 found: NO (tried pkgconfig and cmake)
Run-time dependency wingpanel found: YES 3.0.3
Run-time dependency json-glib-1.0 found: YES 1.6.6
Run-time dependency granite found: YES 6.2.0
Run-time dependency indicator3-0.4 found: YES 12.10.2
Configuring Constants.vala using configuration
Build targets in project: 1

wingpanel-indicator-ayatana undefined

  User defined options
    prefix: /usr

Found ninja-1.10.1 at /usr/bin/ninja
```
```
username@pc:~/wingpanel-community-indicators$ cd build
username@pc:~/wingpanel-community-indicators/build$ ninja -d explain
ninja explain: deps for 'libayatana.so.p/meson-generated_src_Indicator.c.o' are missing
ninja explain: output libayatana.so.p/src/Indicator.c doesn't exist
ninja explain: libayatana.so.p/src/Indicator.c is dirty
ninja explain: libayatana.so.p/meson-generated_src_Indicator.c.o is dirty
ninja explain: deps for 'libayatana.so.p/meson-generated_src_MetaIndicator.c.o' are missing
ninja explain: libayatana.so.p/src/MetaIndicator.c is dirty
ninja explain: libayatana.so.p/meson-generated_src_MetaIndicator.c.o is dirty
ninja explain: deps for 'libayatana.so.p/meson-generated_src_NamarupaMetaIndicator.c.o' are missing
ninja explain: libayatana.so.p/src/NamarupaMetaIndicator.c is dirty
ninja explain: libayatana.so.p/meson-generated_src_NamarupaMetaIndicator.c.o is dirty
ninja explain: deps for 'libayatana.so.p/meson-generated_src_IndicatorButton.c.o' are missing
ninja explain: libayatana.so.p/src/IndicatorButton.c is dirty
ninja explain: libayatana.so.p/meson-generated_src_IndicatorButton.c.o is dirty
ninja explain: deps for 'libayatana.so.p/meson-generated_src_IndicatorFactory.c.o' are missing
ninja explain: libayatana.so.p/src/IndicatorFactory.c is dirty
ninja explain: libayatana.so.p/meson-generated_src_IndicatorFactory.c.o is dirty
ninja explain: deps for 'libayatana.so.p/meson-generated_src_IndicatorIface.c.o' are missing
ninja explain: libayatana.so.p/src/IndicatorIface.c is dirty
ninja explain: libayatana.so.p/meson-generated_src_IndicatorIface.c.o is dirty
ninja explain: deps for 'libayatana.so.p/meson-generated_src_IndicatorLoader.c.o' are missing
ninja explain: libayatana.so.p/src/IndicatorLoader.c is dirty
ninja explain: libayatana.so.p/meson-generated_src_IndicatorLoader.c.o is dirty
ninja explain: deps for 'libayatana.so.p/meson-generated_src_IndicatorObject.c.o' are missing
ninja explain: libayatana.so.p/src/IndicatorObject.c is dirty
ninja explain: libayatana.so.p/meson-generated_src_IndicatorObject.c.o is dirty
ninja explain: deps for 'libayatana.so.p/meson-generated_src_SubMenuButton.c.o' are missing
ninja explain: libayatana.so.p/src/SubMenuButton.c is dirty
ninja explain: libayatana.so.p/meson-generated_src_SubMenuButton.c.o is dirty
ninja explain: deps for 'libayatana.so.p/meson-generated_src_Settings.c.o' are missing
ninja explain: libayatana.so.p/src/Settings.c is dirty
ninja explain: libayatana.so.p/meson-generated_src_Settings.c.o is dirty
ninja explain: deps for 'libayatana.so.p/meson-generated_build_Constants.c.o' are missing
ninja explain: libayatana.so.p/build/Constants.c is dirty
ninja explain: libayatana.so.p/meson-generated_build_Constants.c.o is dirty
ninja explain: libayatana.so is dirty
[1/13] Compiling Vala source ../src/Indicator.val...nuButton.vala ../src/Settings.vala Constants.vala
../src/IndicatorButton.vala:27.5-27.23: warning: AyatanaCompatibility.IndicatorButton.name hides inherited field `Gtk.Widget.name'. Use the `new' keyword if hiding was intentional
   27 |     private string name;
      |     ^~~~~~~~~~~~~~~~~~~ 
../src/Settings.vala:256.9-256.46: warning: unhandled error `GLib.Error'
  256 |         parser.load_from_data (jsonString, -1);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ 
../src/Settings.vala:241.9-241.46: warning: unhandled error `GLib.Error'
  241 |         parser.load_from_data (jsonString, -1);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ 
../src/Settings.vala:69.13-69.54: warning: unhandled error `GLib.Error'
   69 |             settings_Dir.make_directory_with_parents();
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ 
../src/Settings.vala:73.13-73.61: warning: unhandled error `GLib.Error'
   73 |             settings_Images_Dir.make_directory_with_parents();
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ 
../src/Settings.vala:93.19-95.9: warning: unhandled error `GLib.Error'
   93 |         monitor = settings_File.monitor ( //to track directory use .monitor_directory
      |                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   94 |             GLib.FileMonitorFlags.NONE
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   95 |         );
      | ~~~~~~~~~ 
../src/Settings.vala:128.22-128.38: warning: unhandled error `GLib.Error'
  128 |             pixbuf = info.load_icon ();
      |                      ^~~~~~~~~~~~~~~~~ 
../src/Settings.vala:133.13-133.54: warning: unhandled error `GLib.Error'
  133 |             pixbuf.save (save_file.get_path (), "png");
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ 
../src/NamarupaMetaIndicator.vala:162.5-162.47: warning: Method `AyatanaCompatibility.NamarupaMetaIndicator.get_restrictions_from_file' never used
  162 |     private string[] get_restrictions_from_file (File file) {
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~              
../src/NamarupaMetaIndicator.vala:186.10-186.12: warning: Local variable `msg' declared but never used
  186 | 		string msg = cpt.to_string () + " item(s)\n";
      | 		       ^~~                                   
Compilation succeeded - 10 warning(s)
[13/13] Linking target libayatana.so
```

**Comment by Wolf117Warrior** [comment]
I get the same error.
The icons do not appear.
![image](https://user-images.githubusercontent.com/9154175/216567761-501fd23d-d39e-498b-9cca-5440c46b2868.png)
I have the element.io application running in the background but no icons are present.
The installation was done with meson and then with the deb package.

**Comment by MvBonin** [comment]
Hi
I'll work on the warnings, when I have time. I started using fedora much more but i upgraded my elementaryos to 7 now, so I can look into it.



> The icons do not appear.  I have the element.io application running in the background but no icons are present. The installation was done with meson and then with the deb package.

Please try installing gir1.2-appindicator3-0.1 with 

sudo apt install gir1.2-appindicator3-0.1

and reboot.
I used NordIndicator to test it and it worked after installing gir1.2-appindicator3-0.1.
Please let me know, if this resolves your issue for now.
![Screenshot from 2023-02-15 16-33-04](https://user-images.githubusercontent.com/41330432/219075808-53ef91c7-e170-46e1-a8bc-4e92f59d31a8.png)

**Comment by kalligator** [comment]
@MvBonin Are indicators on Pantheon Fedora possible?

**Comment by MvBonin** [comment]
@kalligator I didn't try that yet..

**Comment by MvBonin** [comment]
Since the compilation works even with the warnings for now and this 

> I get the same error. The icons do not appear. ![image](https://user-images.githubusercontent.com/9154175/216567761-501fd23d-d39e-498b-9cca-5440c46b2868.png) I have the element.io application running in the background but no icons are present. The installation was done with meson and then with the deb package.

can be fixed by installing additional dependencies, 
<pre>sudo apt install gir1.2-appindicator3-0.1</pre>
I'll close the issue for now and added the dependency to the Readme in [b9baee5](https://github.com/MvBonin/wingpanel-community-indicators/commit/b9baee5adeb3f78c7488662cf94b1e746a8510e3).

**Comment by kalligator** [comment]
It's not working properly. 
Try an app that reads mouse wheel scrolling like [radiotray-ng](https://github.com/ebruck/radiotray-ng) and you'll see the custom indicators get vanished.

**Comment by MvBonin** [comment]
> It's not working properly. Try an app that reads mouse wheel scrolling like [radiotray-ng](https://github.com/ebruck/radiotray-ng) and you'll see the custom indicators get vanished as soon as you scroll on it (so as to adjust the volume).

Oh ok. Thats interesting! But since the Indicators appear first, I opened a different issue for this #3

**Comment by MvBonin** [comment]
Fixed warnings in 2d18b68de6a309b713409b454a98bb8a7b8c7185

**Comment by Wolf117Warrior** [comment]
> Hi I'll work on the warnings, when I have time. I started using fedora much more but i upgraded my elementaryos to 7 now, so I can look into it.
> 
> > The icons do not appear.  I have the element.io application running in the background but no icons are present. The installation was done with meson and then with the deb package.
> 
> @Wolf117Warrior Please try installing gir1.2-appindicator3-0.1 with
> 
> sudo apt install gir1.2-appindicator3-0.1
> 
> and reboot. I used NordIndicator to test it and it worked after installing gir1.2-appindicator3-0.1. Please let me know, if this resolves your issue for now. ![Screenshot from 2023-02-15 16-33-04](https://user-images.githubusercontent.com/41330432/219075808-53ef91c7-e170-46e1-a8bc-4e92f59d31a8.png)

Sorry for the delay in response.
I do have the icons that appear, no change before or after installing the package. So, I guess more a problem related to the element.io application.

![ayatana](https://user-images.githubusercontent.com/9154175/222376618-7d2e7d80-6293-47e2-87b2-f42223c687fe.png)
